src/hakunapi-jsonfg/src/main/java/fi/nls/hakunapi/geojson/hakuna/JSONFGFeatureCollectionWriter.java

```diff
@@ -0,0 +1,233 @@
+
+		// type
+		json.writeFieldName(HakunaGeoJSON.TYPE);
+		json.writeStringUnsafe(HakunaGeoJSON.FEATURE_COLLECTION, 0, HakunaGeoJSON.FEATURE_COLLECTION.length);
+
+		json.writeStringField(FEATURE_TYPE, ftName);
+
+		if (geometryDimension != null) {
+			json.writeFieldName(GEOMETRY_DIMENSION);
+			json.writeNumber(geometryDimension);
+		}
+		if (ftJSONFGSchema != null) {
+			json.writeStringField(FEATURE_SCHEMA, ftJSONFGSchema);
+		}
+
+		if (!isWgs84) {
+			String coordRefSys = CrsUtil.toUri(outputSrid);
+			json.writeStringField(COORD_REF_SYS, coordRefSys);
+		}
+
+		json.writeFieldName(HakunaGeoJSON.FEATURES);
+		json.writeStartArray();
+	}
+
+	@Override
+	public void endFeatureCollection() throws IOException {
+		// close "features":[ ]
+		json.writeEndArray();
+		collectionFt = null;
+	}
+
+	@Override
+	public void end(boolean timeStamp, List<Link> links, int numberReturned) throws IOException {
+		if (timeStamp) {
+			writeTimeStamp();
+		}
+		writeLinks(links);
+		writeNumberReturned(numberReturned);
+	}
+
+	@Override
+	public void startFeature(String fid) throws IOException {
+		startFeature();
+		json.writeString(fid);
+	}
+
+	@Override
+	public void startFeature(long fid) throws IOException {
+		startFeature();
+		json.writeNumber(fid);
+	}
+
+	private void startFeature() throws IOException {
+		date = null;
+		timestamp = null;
+		cachedGeometry = null;
+		geometryCached = false;
+		json.writeStartObject();
+		json.writeFieldName(HakunaGeoJSON.TYPE);
+		json.writeStringUnsafe(HakunaGeoJSON.FEATURE, 0, HakunaGeoJSON.FEATURE.length);
+		json.writeFieldName(HakunaGeoJSON.ID);
+
+	}
+
+	@Override
+	public void writeProperty(String name, Instant value) throws Exception {
+		super.writeProperty(name, value);
+
+		// TODO what when why multiple temporal?
+		if (timestamp == null && value != null) {
+			timestamp = value;
+		}
+	}
+
+	@Override
+	public void writeProperty(String name, LocalDate value) throws Exception {
+		super.writeProperty(name, value);
+
+		// TODO what when why multiple temporal?
+		if (date == null && value != null) {
+			date = value;
+		}
+	}
+
+	@Override
+	public void writeGeometry(String name, HakunaGeometry geometry) throws Exception {
+		// force cache to enable quirks
+		geometryCached = true;
+		cachedGeometry = geometry;
+	}
+
+	@Override
+	public void endFeature() throws Exception {
+		closeProperties();
+
+		// TODO: pre reprojected geoms
+		if (geometryCached && cachedGeometry != null) {
+			// PLACE
+			cachedGeometry.write(placeJson);
+
+			// GEOMETRY WGS84
+			// TOWGS84
+			Geometry wgs84Geometry = ProjectionHelper.reproject(cachedGeometry.toJTSGeometry().copy(), outputWgs84Proj);
+			HakunaGeometryJTS jts = new HakunaGeometryJTS(wgs84Geometry);
+			jts.write(geometryJson);
+
+			// cleanup
+			geometryCached = false;
+			cachedGeometry = null;
+
+		} else {
+			// TODO check if this is required
+			json.writeFieldName(GEOMETRY);
+			json.writeNull();
+
+		}
+
+		// TODO what when why multiple temporal?
```

Comment:
In `startFeatureCollection(FeatureType ft, String layername)` 
use `ft.getDatetimeProperties()` to get the datetime properties associated with time, cache their (public) names and on each `writeProperty(Instant | LocalDate)` check if the name is one of the configured ones and omit writing those out right away. Core 1 `?datetime=start/end` also gets compared against these properties.

Currently there's no way to tell if two configured time properties should be considered to form a time range, e.g.:
```
collections.my_collection.time=version_start_time,version_end_time
collections.my_collection.time.exclusive=version_end_time
```
but I think it's a safe assumption to make.
```
int size = ft.getDatetimeProperties().size();
if (size == 1) {
    // expect date/timestamp (check the property type for which to expect)
} else if (size == 2) {
  // expect and write out interval
}

Reply:
supports only one datetimeproperty at the moment